srcjs/shinyapp.js

```diff
@@ -696,6 +696,12 @@ var ShinyApp = function() {
     });
   });
 
+  addMessageHandler('frozen', function(message) {
+    for (let i = 0; i < message.ids.length; i++) {
+      exports.forgetLastInputValue(message.ids[i]);
```

Comment:
Will this handle inputs that have a type, like `x:num` (at least, I think that's what it is)?

Reply:
Yes, these are handled. You can't call `forgetLastInput("x:num")` but you can call `forgetLastInput("x")` even if `x` has a type of `"num"`.